torchgeo/datasets/biomassters.py

```diff
@@ -0,0 +1,344 @@
+        """
+        self.root = root
+
+        assert (
+            split in self.valid_splits
+        ), f"Please choose one of the valid splits {self.valid_splits}."
+        self.split = split
+
+        assert set(sensors).issubset(
+            set(self.valid_sensors)
+        ), f"Please choose a subset of valid sensors {self.valid_sensors}."
+        self.sensors = sensors
+        self.as_time_series = as_time_series
+        self.download = download
+        self.checksum = checksum
+
+        try:
+            import pandas as pd  # noqa: F401
+        except ImportError:
+            raise ImportError(
+                "pandas is not installed and is required to use this dataset"
+            )
+
+        self._verify()
+
+        # open metadata csv files
+        self.df = pd.read_csv(os.path.join(self.root, self.metadata_filename))
+
+        # filter sensors
+        self.df = self.df[self.df["satellite"].isin(self.sensors)]
+
+        # filter split
+        self.df = self.df[self.df["split"] == self.split]
+
+        # generate numerical month from filename since first month is September
+        # and has numerical index of 0
+        self.df["num_month"] = (
+            self.df["filename"]
+            .str.split("_", expand=True)[2]
+            .str.split(".", expand=True)[0]
+            .astype(int)
+        )
+
+        # set dataframe index depending on the task for easier indexing
+        if self.as_time_series:
+            self.df["num_index"] = self.df.groupby(["chip_id"]).ngroup()
+        else:
+            filter_df = (
+                self.df.groupby(["chip_id", "month"])["satellite"].count().reset_index()
+            )
+            filter_df = filter_df[filter_df["satellite"] == len(self.sensors)].drop(
+                "satellite", axis=1
+            )
+            # guarantee that each sample has corresponding number of images available
+            self.df = self.df.merge(filter_df, on=["chip_id", "month"], how="inner")
+
+            self.df["num_index"] = self.df.groupby(["chip_id", "month"]).ngroup()
+
+    def __getitem__(self, index: int) -> dict[str, Any]:
+        """Return an index within the dataset.
+
+        Args:
+            index: index to return
+
+        Returns:
+            data and labels at that index
+
+        Raises:
+            IndexError: if index is out of range of the dataset
+        """
+        sample_df = self.df[self.df["num_index"] == index].copy()
+
+        # sort by satellite and month to return correct order
+        sample_df.sort_values(
+            by=["satellite", "num_month"], inplace=True, ascending=True
+        )
+
+        filepaths = sample_df["filename"].tolist()
+        sample: dict[str, Tensor] = {}
+        for sens in self.sensors:
+            sens_filepaths = [fp for fp in filepaths if sens in fp]
+            sample[f"image_{sens}"] = self._load_input(sens_filepaths)
```

Comment:
This relates to #985. I would love to be able to avoid custom keys in dicts (it prevents us from using these datasets with our trainers or type hint them). I'm fine with this for now, but we should think of whether or not it's possible to keep these as standard names.